File: testdata/fixtures/c/CWE-611.c
/* CWE-611: XXE */
#include <libxml/parser.h>
#include <expat.h>
void vuln(const char *user, const char *buf, XML_Parser p) {
    /* ruleid: c-xxe */
    xmlReadFile(user, NULL);
    /* ruleid: c-xxe */
    xmlReadMemory(buf, 100, user, NULL, 0);
    /* ruleid: c-xxe */
    XML_Parse(p, user, 100);
    /* ok: c-xxe */
    xmlReadFile("/tmp/fixed.xml", NULL);
    /* ok: c-xxe */
    XML_Parse(p, "<root>hi</root>", 14);
}

File: testdata/fixtures/php/CWE-611.php
<?php
// CWE-611: XXE
function vuln($xml) {
    // ruleid: php-xxe
    $a = simplexml_load_string($xml);
    // ruleid: php-xxe
    $b = simplexml_load_file($xml);
    // ruleid: php-xxe
    libxml_disable_entity_loader(false);
}
function safe() {
    // ok: php-xxe
    $a = simplexml_load_string("<root>fixed</root>");
    // ok: php-xxe
    $b = simplexml_load_file("/tmp/fixed.xml");
    // ok: php-xxe
    libxml_disable_entity_loader(true);
}
